### ISSUE TITLE
[Wiko Y60] when downloading music a good chunk of them don't actually get downloaded even if the download section says it is done

### ISSUE THREAD

**Issue by milkywilk** [jmshrv/finamp#384]
![Screenshot_20230115-124959](https://user-images.githubusercontent.com/122728299/212539180-e02cc783-5512-4d1c-ab44-7c1f271930ec.png)
![Screenshot_20230115-124935](https://user-images.githubusercontent.com/122728299/212539182-28520f6d-4416-45dd-aa62-b2c64102214d.png)

the songs not downloaded do not have the "downloaded icon", there is no way to force redownload them besides of deleting the whole album off the phone and redownloading it, in which case the same thing could happen again.
i do not get errors when it comes to this, it simply doesn't download them

i am assuming this is a bug.

**Comment by jmshrv** [comment]
That's weird. Could you try downloading, and send logs after it's "done"?

**Comment by milkywilk** [comment]
> That's weird. Could you try downloading, and send logs after it's "done"?

it seems to be an issue with bulk downloading multiple albums at the same time
when i try to download multiple albums at the same time some songs get left
also it was using local storage as i haven't gotten an SD card at the time
it works fine if i download an album or 2, wait for it to finish and then queue in the next one
[finamp-logs.txt](https://github.com/jmshrv/finamp/files/10423576/finamp-logs.txt)
i spotted 3 songs that didn't get downloaded this time
in kanye west - graduation:
06 - Can't Tell me Nothing
09 - Flashing Lights feat. Dwele
12 - Homecoming

**Comment by jmshrv** [comment]
Hmm yeah downloading loads of stuff at once can cause issues (afaik this is related to the app getting killed by the OS), it might be fixed when I get around to #213

**Comment by cbeauhilton** [comment]
Just wanted to +1 - I'm having the same issue as @milkywilk , maybe a bit worse (not getting through even a single album download with every track downloaded, most of the time some are missing), though I expect part of the issue is that my upload from the home server is very slow atm. Would make sense if it's an autokill issue with iOS.

**Comment by jmshrv** [comment]
Yeah in that case it's probably iOS being awkward. If you can, keep Finamp open while downloading

**Comment by CPU-Blanc** [comment]
+1 on this issues too. Tried to download a playlist of ~500 songs about 8 times now. Each time it says the download is complete but a bunch of songs are missing. On Android here so not limited to iOS. Tried disabling the screen look and leaving the app open but that doesn't help.

**Comment by jmshrv** [comment]
The app may be overloading the network at some point (not to the point of failure or anything, just sending too many requests). I should improve the infrastructure of this bit of the app, maybe adding a progress bar for "adding downloads" instead of just hoping it all goes through.

**Comment by csaavedra** [comment]
Still happening.

**Comment by Chaphasilor** [comment]
@milkywilk @CPU-Blanc the download system has been improved as part of the redesign beta. You can check it out here: https://github.com/jmshrv/finamp/releases/tag/0.9.2-beta

I'll close this issue for now since it should be resolved.

**Comment by CPU-Blanc** [comment]
@Chaphasilor Tried the beta, seems to be working a lot smoother. Thank you :+1:
